content/en/docs/ops/ambient/upgrade/helm-upgrade/index.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Upgrade with Helm
+description: How to upgrade Ambient Mesh with Helm.
+weight: 5
+owner: istio/wg-environments-maintainers
+test: yes
+---
+
+Follow this guide to upgrade and configure ambient mesh using
+[Helm](https://helm.sh/docs/).  This guide assumes you have already performed an ambient mesh
+[installation with Helm](/docs/ops/ambient/install/helm-installation/) with a previous minor or patch version of Istio.
+
+{{< boilerplate ambient-alpha-warning >}}
+
+{{< warning >}}
+`Ambient` does not currently support canary upgrades of the dataplane. This means that ambient mesh dataplane upgrades will disrupt application pod traffic. Node cordoning and blue/green node pools are recommended to mitigate blast radius risk during production upgrades. See your Kubernetes provider documentation for details.
+{{< /warning >}}
+
+{{< boilerplate helm-preamble >}}
+
+{{< boilerplate helm-prereqs >}}
+
+## Prerequisites
+
+1. Install ambient mesh with Helm by following [Install with Helm](/docs/ops/ambient/install/helm-installation/) and meeting all prerequisites in that guide.
+
+1. Update the Helm repository:
+
+    {{< text syntax=bash snip_id=update_helm >}}
+    $ helm repo update istio
+    {{< /text >}}
+
+## In-place upgrade
+
+You can perform an in place upgrade of Istio in your cluster using the Helm
+upgrade workflow.
+
+Before upgrading Istio, it is recommended to run the `istioctl x precheck` command to make sure the upgrade is compatible with your environment.
+
+{{< text syntax=bash snip_id=istioctl_precheck >}}
+$ istioctl x precheck
+✔ No issues found when checking the cluster. Istio is safe to install or upgrade!
+  To get started, check out <https://istio.io/latest/docs/setup/getting-started/>
+{{< /text >}}
+
+{{< warning >}}
+[Helm does not upgrade or delete CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations) when performing an upgrade. Because of this restriction, an additional step is required when upgrading Istio with Helm.
+{{< /warning >}}
+
+### Manually upgrading the CRDs and Istio base chart
+
+1. Upgrade the Kubernetes custom resource definitions ({{< gloss >}}CRDs{{</ gloss >}}):
+
+    {{< text syntax=bash snip_id=manual_crd_upgrade >}}
+    $ kubectl apply -f manifests/charts/base/crds
+    {{< /text >}}
+
+1. Upgrade the Istio base chart:
+
+    {{< text syntax=bash snip_id=upgrade_base >}}
+    $ helm upgrade istio-base manifests/charts/base -n istio-system --skip-crds
+    {{< /text >}}
+
+### Upgrading the Istio discovery Component
+
+Istiod is the control plane component that manages and configures the proxies to route traffic within the ambient mesh.
+
+{{< text syntax=bash snip_id=upgrade_istiod >}}
+$ helm upgrade istiod istio/istiod -n istio-system
+{{< /text >}}
+
```

Comment:
Does it make sense to upgrade istio-cni here after control plane? or before control plane?  Feels it is part of control plane and we are asking users to update it after dataplane atm.

Reply:
It does not matter (or rather it should not matter) as long as the skew is respected as per what @dhawton pointed out.

The ordering logic here is

- Required steps
- Optional steps

It is, strictly speaking, optional to upgrade the CNI, so it goes after the CP (which is not optional) for that reason. 

Would be good to have it both ways in docs tests to catch issues anyway.

Reply:
Note we have some existing guideline here - https://istio.io/latest/docs/setup/additional-setup/cni/#operation-details

Reply:
iirc, our recommendation is to upgrade istio-cni before or after istiod.

Reply:
For sidecar it matters, and I don't want to change sidecar ordering reqs.

For ambient, it does not, we have no direct control plane dependency in CNI. So I don't want to convey to people that there is a preferred ordering, because there isn't. Both should work and be supported.

Reply:
Understood it may not matter but we may not have tested all these permutations in the past and I'm a bit cautious on recommending new orderings with unnecessary benefits to users.

Reply:
For inpod, the recommendation will be to upgrade `istio-cni` when `ztunnel` is upgraded, if it is upgraded at all. Istiod isn't a factor.